parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -362,6 +364,11 @@ public ReadBuilder createReaderFunc(Function<MessageType, ParquetValueReader<?>>
       return this;
     }
 
+    public ReadBuilder createBatchedReaderFunc(Function<MessageType, VectorizedReader<?>> func) {
```

Comment:
Can you add preconditions to validate that the other reader func is not set, and one in the other method for when this one is already set? That way we throw an `IllegalArgumentException` when both are called and we don't have to worry about what to do when the user configures two object models.